### PR TITLE
Add basic UI for Forced password reset. Refer the wireframe [https://…

### PR DESCRIPTION
**Pull request by DMHP**
…github.com/wso2-dev-ux/product-is/blob/master/Wireframes/admin-portal/v3/3.32%20%20Reset%20password%20with%20offline%20OTP%20-%20password%20generated.png]

### CODE REVIEW

File: portal/apps/org.wso2.carbon.iam.admin-portal.app/src/main/pages/recovery/forced-password-reset.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+function onGet(env) {
+    var session = getSession();
+    if (!session || !session.getUser()) {
```
Comment (kasunbg) [inline portal/apps/org.wso2.carbon.iam.admin-portal.app/src/main/pages/recovery/forced-password-reset.js]: This is unnecessary.. Use {{secured}} handlebar in your hbs file to redirect unauthorized users to the login page.

File: portal/apps/org.wso2.carbon.iam.admin-portal.app/src/main/lang/en_US.properties [hunk 2]
```diff
@@ -32,3 +32,9 @@ users.add.select.with.otp.and.askPassword=With OTP and ask password from user
 users.add.select.with.email.or.phone=With email/phone verification
 users.add.password=Password
 users.confirm.password=Confirm Password
+recovery.reset.password=Reset Password
+recovery.reset.selectMethod=Select password reset method
+recovery.reset.select.with.offline.OTP=With offline OTP
```
Comment (kasunbg) [inline portal/apps/org.wso2.carbon.iam.admin-portal.app/src/main/lang/en_US.properties]: Can we improve the text to say 'Via Offline OTP'

File: portal/apps/org.wso2.carbon.iam.admin-portal.app/src/main/lang/en_US.properties [hunk 3]
```diff
@@ -32,3 +32,9 @@ users.add.select.with.otp.and.askPassword=With OTP and ask password from user
...
+recovery.reset.select.with.email.OTP=With email OTP
+recovery.reset.select.with.reset.link=With reset link
```
Comment (kasunbg) [inline portal/apps/org.wso2.carbon.iam.admin-portal.app/src/main/lang/en_US.properties]: Same as before.

File: portal/client-services/org.wso2.is.portal.user.client.api/src/main/java/org/wso2/is/portal/user/client/api/RecoveryMgtServiceImpl.java [hunk 4]
```diff
@@ -268,6 +269,16 @@ public boolean verifyUsername(Map<String, String> userClaims) throws IdentityRec
         return getNotificationUsernameRecoveryManager().verifyUsername(claims);
     }
 
+    @Override
+    public void insertOTPValue(String uniqueUserId, String otp) throws UserPortalUIException {
```
Comment (kasunbg) [inline portal/client-services/org.wso2.is.portal.user.client.api/src/main/java/org/wso2/is/portal/user/client/api/RecoveryMgtServiceImpl.java]: Can we add docs for this method? The method name is a bit unclear.

File: portal/client-services/org.wso2.is.portal.user.client.api/src/main/java/org/wso2/is/portal/user/client/api/RecoveryMgtService.java [hunk 5]
```diff
@@ -89,4 +89,11 @@ ChallengeQuestionsResponse verifyUserChallengeAnswers(String code,
      */
     boolean verifyUsername(Map<String, String> userClaims) throws IdentityRecoveryException;
 
+    /**
+     *
+     * @param userUnqueId selected user id
+     * @param otp generated one time password
+     */
+    void insertOTPValue(String userUnqueId, String otp) throws UserPortalUIException;
```
Comment (kasunbg) [inline portal/client-services/org.wso2.is.portal.user.client.api/src/main/java/org/wso2/is/portal/user/client/api/RecoveryMgtService.java]: Typo in the first parameter name

File: portal/client-services/org.wso2.is.portal.user.client.api/src/main/java/org/wso2/is/portal/user/client/api/micro/IdentityStoreClientMicroService.java [hunk 6]
```diff
@@ -34,7 +37,6 @@
  * This is the micro service implementation for identity store client services.
  */
 public class IdentityStoreClientMicroService implements Microservice {
```
Comment (GayanM) [inline portal/client-services/org.wso2.is.portal.user.client.api/src/main/java/org/wso2/is/portal/user/client/api/micro/IdentityStoreClientMicroService.java]: why micro service to generate OTP?

File: portal/client-services/org.wso2.is.portal.user.client.api/src/main/java/org/wso2/is/portal/user/client/api/RecoveryMgtServiceImpl.java [hunk 7]
```diff
@@ -268,6 +269,16 @@ public boolean verifyUsername(Map<String, String> userClaims) throws IdentityRec
         return getNotificationUsernameRecoveryManager().verifyUsername(claims);
     }
 
+    @Override
```
Comment (GayanM) [inline portal/client-services/org.wso2.is.portal.user.client.api/src/main/java/org/wso2/is/portal/user/client/api/RecoveryMgtServiceImpl.java]: java doc comments ?

File: portal/client-services/org.wso2.is.portal.user.client.api/src/main/java/org/wso2/is/portal/user/client/api/micro/IdentityStoreClientMicroService.java (hunk 6 above)
Comment (DMHP) [inline portal/client-services/org.wso2.is.portal.user.client.api/src/main/java/org/wso2/is/portal/user/client/api/micro/IdentityStoreClientMicroService.java]: As Generate Password is not a submit we needed to handle it using an ajax call. So had to use micro service.

File: portal/apps/org.wso2.carbon.iam.admin-portal.app/src/main/pages/recovery/forced-password-reset.hbs [hunk 8]
```diff
@@ -0,0 +1,44 @@
+{{secured}}
+{{layout "privilege"}}
+{{favicon "images/favicon.png" type="image/png"}}
+{{title "Reset Password | " @config.appName}}
```
Comment (denuwanthi) [inline portal/apps/org.wso2.carbon.iam.admin-portal.app/src/main/pages/recovery/forced-password-reset.hbs]: We can internationalize the tiltle. As there can be need to use other languages.
You can do it by 
{{title (i18n "users.add.header") " | " @config.appName}}

**Pull request merged**